erpnext/public/build.json

```diff
@@ -30,6 +30,7 @@
         "public/js/payment/payment_details.html",
         "public/js/templates/item_selector.html",
         "public/js/utils/item_selector.js",
+        "public/js/utils/serial_no_batch_selector.js",
```

Comment:
don't add this here... this will add to the initial load.

use `frappe.require` in the files you need just JS. The file will be loaded in background by the time the user comes to the item selection